Promact.Oauth.Server/src/Promact.Oauth.Server/Repository/UserRepository/UserRepository.cs

```diff
@@ -783,6 +645,87 @@ private string GetRandomString()
             return sb.ToString();
         }
 
+        /// <summary>
+        /// Calculat casual leava and sick leave from the date of joining
+        /// </summary>
+        /// <param name="dateTime"></param>
+        /// <returns></returns>
+        private LeaveCalculator CalculateAllowedLeaves(DateTime dateTime)
```

Comment:
Why is this method private? private methods are not unit testable. Additionally, there is no line-by-line documentation to explain the flow of calculations.